packages/java/parser-jvm-plugin-nonnull-kotlin/src/main/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/KotlinNullabilityPlugin.kt

```diff
@@ -0,0 +1,258 @@
+            is MethodNode -> createKMethodNode(node, parentPath)
+            is MethodParameterNode -> KMethodParameterNode(node.source, node.target,
+                    (parentPath.node as KMethodNode).kFunction.parameters
+                        .first { it.kind == KParameter.Kind.VALUE && it.name == node.source.name })
+            is TypedNode -> resolveTypedNode(node, parentPath)
+            is EntityNode ->
+                KEntityNode(node.source, node.target as ObjectSchema, (node.source.get() as Class<*>).kotlin)
+            is PropertyNode -> {
+                val kProperty = (parentPath.node as? KEntityNode)?.kClass?.memberProperties
+                    ?.firstOrNull { it.name == node.source.name }
+                if (kProperty != null) {
+                    KPropertyNode(node.source, node.target, kProperty)
+                } else {
+                    // If the property is not found, it possibly can be a property defined using getter/setter methods
+                    // then leave it unchanged (will be handled in exit)
+                    node
+                }
+            }
+            else -> super.resolve(node, parentPath)
+        }
+    }
+
+    private fun resolveTypedNode(node: TypedNode, parentPath: NodePath<*>): Node<*, *> {
+        return when (node.type) {
+            is TypeArgumentModel -> resolveTypeArgumentModel(node as TypeSignatureNode, parentPath)
+            is ClassRefSignatureModel, is BaseSignatureModel, is TypeVariableModel -> resolveTypeSignature(node, parentPath)
+            else -> node as Node<*, *>
+        }
+    }
+
+    /**
+     * This handles these cases:
+     * - generic argument of method return type (and their nested generic arguments)
+     * - generic argument of method parameter type (and their nested generic arguments)
+     *
+     * This does not handle cases:
+     * - generic arguments in class definition, e.g. as 'Long' in:
+     *      class Person: AbstractEntity<Long>()
+     */
+    private fun resolveTypeArgumentModel(node: TypeSignatureNode, parentPath: NodePath<*>): Node<*, *> {
+        return when (parentPath.node) {
+            is KTypeSignatureNode -> {
+                val parentType = (parentPath.node as KTypeSignatureNode).kType
+                // For maps, always use index 1 (ignore key type); otherwise, use the node's position.
+                val position = if (parentType.toString().startsWith("kotlin.collections.Map<")) 1
+                else node.position
+                KTypeSignatureNode(
+                    node.type, node.target, node.annotations, node.position,
+                    parentType.arguments[position].type!!
+                )
+            }
+            else -> node
+        }
+    }
+
+    private fun resolveTypeSignature(node: TypedNode, parentPath: NodePath<*>): Node<*, *> {
+        return when (parentPath.node) {
```

Comment:
You can skip the `{ return }` and use `when (val parent = parentPath.node)` to replace `parentPath.node` in the block